### PR TITLE
authz: remove alpha RBAC CRD

### PR DESCRIPTION
**Pull request by yangminzhu**
The API is removed in https://github.com/istio/api/pull/1427.

#23668

### CODE REVIEW

File: tests/integration/galley/analyze_test.go [hunk 1]
```diff
@@ -76,15 +77,10 @@ func TestFileOnly(t *testing.T) {
...
 
-			// Validation error if we have a service role binding without a service role
-			output, err := istioctlSafe(t, istioCtl, ns.Name(), false, serviceRoleBindingFile)
+			// Validation error if we have a gateway with invalid selector
+			output, err := istioctlSafe(t, istioCtl, ns.Name(), false, gatewayFile)
 			expectMessages(t, g, output, msg.ReferencedResourceNotFound)
 			g.Expect(err).To(BeIdenticalTo(analyzerFoundIssuesError))
-
-			// Error goes away if we include both the binding and its role
-			output, err = istioctlSafe(t, istioCtl, ns.Name(), false, serviceRoleBindingFile, serviceRoleFile)
```
Comment (howardjohn) [inline tests/integration/galley/analyze_test.go:85]: this is changing the test, we no longer verify we can fix errors

**Review by howardjohn — APPROVED** [review]

File: tests/integration/galley/analyze_test.go (hunk 1 above)
Comment (yangminzhu) [reply tests/integration/galley/analyze_test.go:85]: thanks, fixed.

**Review by incfly — APPROVED** [review]

**Comment by yangminzhu** [comment]
@istio/wg-docs-maintainers @istio/wg-policies-and-telemetry-maintainers @istio/wg-user-experience-maintainers could you take a look? thanks.

**Review by ericvn — APPROVED** [review]

**Review by mandarjog — APPROVED** [review]
lgtm

**Review by jasonwzm — APPROVED** [review]

**Pull request merged**